azurelinuxagent/common/osutil/redhat.py

```diff
@@ -140,6 +139,30 @@ def get_nm_controlled(self):
 
         return True
 
+    def check_and_recover_nic_state(self, ifname):
+        filepath = "/sys/class/net/{0}/operstate".format(ifname)
+        try:
+            if not os.path.isfile(filepath):
+                logger.warn("Unable to determine primary network interface state, because state file does not exist: "
+                            "{0}".format(filepath))
```

Comment:
we should send telemetry about those errors so we can fix them, should they ever happen